client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import DomainPicker from '@automattic/domain-picker';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
+import { DomainSuggestions } from '@automattic/data-stores';
+import {
+	MARKETPLACE_FLOW_ID,
+	ANALYTICS_UI_LOCATON_MARKETPLACE_DOMAIN_SELECTION,
+} from 'calypso/my-sites/plugins/marketplace/constants';
+import { Button } from '@wordpress/components';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+import './styles.scss';
+import { getProductsList, isProductsListFetching } from 'calypso/state/products-list/selectors';
+
+type DomainSuggestion = DomainSuggestions.DomainSuggestion;
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+
+function Header() {
+	return (
+		<div className="marketplace-domain-upsell__header domains__header">
+			<h1 className="marketplace-domain-upsell__header marketplace-title">Choose a domain</h1>
```

Comment:
Do we really want to use the same `className` on both the parent and the child? Note that this applies to multiple instances in this PR.

In any case, I'd recommend removing any `className`s that we don't need, and leave just the ones that we actually need for styling or functionality.

Reply:
I was getting a lint error whenever I don't use this classname, I thought it might be a convention. Is this not needed then?

Reply:
We definitely don't want to apply the same classname to elements that have a completely different role - this most likely indicates a problem with the organization of our styles.

Perhaps we can try removing any of the unused classes? Then rename all the classnames to fulfill the ESLint rule? Also, note that moving each of the components to separate files will help with this problem.

Reply:
So I have aggresively migrated to emotion while leaving the big picture layout styles in the sass file. Please let me know what you think about this! I am also learning to strike a balance here :)